```
======================================================================
ERROR: testCaseAPIResultJSON (corehq.apps.cloudcare.tests.test_api.CaseAPITest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/corehq/apps/cloudcare/tests/test_api.py", line 31, in setUp
    create_domain(self.domain)
  File "/mnt/corehq/apps/domain/shortcuts.py", line 9, in create_domain
    secure_submissions=False)
  File "/mnt/corehq/apps/domain/models.py", line 553, in get_or_create_with_name
    new_domain.save(**get_safe_write_kwargs())
  File "/mnt/corehq/apps/domain/models.py", line 611, in save
    super(Domain, self).save(**params)
  File "/mnt/corehq/apps/cachehq/mixins.py", line 17, in save
    self.clear_caches()
  File "/mnt/corehq/apps/domain/models.py", line 1060, in clear_caches
    super(Domain, self).clear_caches()
  File "/mnt/corehq/apps/cachehq/mixins.py", line 58, in clear_caches
    super(QuickCachedDocumentMixin, self).clear_caches()
  File "/mnt/corehq/apps/cachehq/mixins.py", line 11, in clear_caches
    invalidate_document(self, couch_db=self.get_db())
  File "/mnt/corehq/apps/cachehq/invalidate.py", line 16, in invalidate_document
    get_cache_pillow(couch_db).change_trigger({
  File "submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py", line 11, in _inner
    return m(*args, **kwargs)
  File "submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py", line 128, in __call__
    value = self.func( *args, **kwargs)
  File "/mnt/corehq/apps/cachehq/invalidate.py", line 7, in get_cache_pillow
    return cacheinvalidate.CacheInvalidatePillow(couch_db=couch_db)
TypeError: __init__() takes at least 3 arguments (2 given)
```

--- a/corehq/apps/cachehq/invalidate.py
+++ b/corehq/apps/cachehq/invalidate.py
@@ -4,7 +4,7 @@ from dimagi.utils.decorators.memoized import memoized
 
 @memoized
 def get_cache_pillow(couch_db):
-    return cacheinvalidate.CacheInvalidatePillow(couch_db=couch_db)
+    return cacheinvalidate.CacheInvalidatePillow(pillow_id='CacheInvalidatePillow', couch_db=couch_db)
 
 
 def invalidate_document(document, couch_db, deleted=False):

--- a/corehq/ex-submodules/pillowtop/tests/test_config.py
+++ b/corehq/ex-submodules/pillowtop/tests/test_config.py
@@ -9,7 +9,7 @@ class PillowConfigTest(SimpleTestCase):
         config = get_pillow_config_from_setting('my-section', class_name)
         self.assertEqual('my-section', config.section)
         self.assertEqual(class_name, config.class_name)
-        self.assertEqual(class_name, config.instance_generator)
+        self.assertEqual(None, config.instance_generator)
 
     def test_from_dict(self):
         pillow_config = {

--- a/corehq/ex-submodules/pillowtop/tests/test_import_pillows.py
+++ b/corehq/ex-submodules/pillowtop/tests/test_import_pillows.py
@@ -43,10 +43,10 @@ class FakeConstructedPillow(ConstructedPillow):
     pass
 
 
-def make_fake_constructed_pillow():
+def make_fake_constructed_pillow(pillow_id):
     fake_dao = MockDocumentStore()
     pillow = FakeConstructedPillow(
-        name='FakeConstructedPillowName',
+        name=pillow_id,
         document_store=fake_dao,
         checkpoint=PillowCheckpoint('fake-constructed-pillow'),
         change_feed=RandomChangeFeed(10),
@@ -98,7 +98,7 @@ class PillowFactoryFunctionTestCase(SimpleTestCase):
 class PillowTestCase(TestCase):
 
     def test_pillow_reset_checkpoint(self):
-        pillow = make_fake_constructed_pillow()
+        pillow = make_fake_constructed_pillow('FakeConstructedPillowName')
         seq_id = '456'
         pillow.set_checkpoint(Change('123', seq_id))
         self.assertEqual(pillow.checkpoint.get_or_create_wrapped().document.sequence, seq_id)

